tests/Doctrine/Tests/ODM/PHPCR/Mapping/ClassMetadataTest.php

```diff
@@ -324,6 +324,90 @@ public function testClassMetadataInstanceSerializationTranslationProperties($cm)
         $this->assertTrue(in_array('translatedField', $cm->translatableFields));
         $this->assertEquals('locale', $cm->localeMapping);
     }
+
+    /**
+     * It should throw an exception if given a child class FQN when the
+     * metadata is for a leaf.
+     *
+     * @expectedException \Doctrine\ODM\PHPCR\Exception\OutOfBoundsException
+     * @expectedExceptionMessage has been mapped as a leaf
+     */
+    public function testAssertValidChildClassesIsLeaf()
+    {
+        $cm = new ClassMetadata('Doctrine\Tests\ODM\PHPCR\Mapping\Person');
+        $childCm = new ClassMetadata('stdClass');
+        $cm->setIsLeaf(true);
+        $cm->assertValidChildClass($childCm);
+    }
+
+    /**
+     * It should return early if the mapped child classes value is an empty array (i.e. any child classes are permitted).
+     */
+    public function testAssertValidChildClassesEmpty()
+    {
+        $cm = new ClassMetadata('Doctrine\Tests\ODM\PHPCR\Mapping\Person');
+        $childCm = new ClassMetadata('stdClass');
+        $cm->setChildClasses(array());
+        $result = $cm->assertValidChildClass($childCm);
+        $this->assertNull($result);
+    }
+
+    /**
+     * It should return early if the given class is an allowed child class.
+     */
+    public function testAssertValidChildClassesAllowed()
+    {
+        $cm = new ClassMetadata('Doctrine\Tests\ODM\PHPCR\Mapping\Person');
+        $cm->setChildClasses(array('stdClass'));
+        $childCm = new ClassMetadata('stdClass');
+        $childCm->initializeReflection(new RuntimeReflectionService());
+        $result = $cm->assertValidChildClass($childCm);
+        $this->assertNull($result);
```

Comment:
Well, there are plans for a void return type in PHP 7.x. So testing null would result in test fails if that RFC is accepted.

And probably personal, for me tests describe the public API and its behaviour. Now the tests seem to tell "it returns null if a valid document is provided". However, the real behaviour is it's throwing exceptions when something is invalid and isn't doing anything if it's valid.